draft-ietf-quic-qpack.md

```diff
@@ -427,16 +427,20 @@ The encoder MUST not set a dynamic table capacity that exceeds this maximum, but
 it can choose to use a lower dynamic table capacity (see
 {{set-dynamic-capacity}}).
 
+For clients using 0-RTT data in HTTP/3, the maximum table capacity is the
+remembered value of the setting. If the remembered value is not present or is 0,
+the server MAY set it to a non-zero value in its SETTINGS frame. If the
```

Comment:
I am not sure if allowing server to accept 0-RTT without remembering the table capacity is a good idea, because it might lead to a connection error.

Instead, can we simply forbid the use of 0-RTT unless the server remembers the value (possibly through the use of session ticket)?

Reply:
I agree with @kazuho.  If the remember value is "not present" on the server side, but it is on the client side, that can cause a problem.

Also, in my mind "the server MAY set it" and "if [...] omits SETTINGS_QPACK_MAX_TABLE_CAPACITY [...], the encoder MUST treat this as [...] error" are inconsistent.

Reply:
I was thinking of a case where the client was attempting 0-RTT from a session which never set SETTINGS_QPACK_MAX_TABLE_CAPACITY (leveraging the default, which is 0), rather than a case where the server had sent one but can't remember.

The 'MAY' refers to when 0-RTT is resumed using 0 or None (implying 0, the default), where the 'MUST treat as error' refers to when 0-RTT is resumed with non-zero.  

But if I confused both of you, then the text needs work. 

> For clients using 0-RTT data in HTTP/3, the server's maximum table capacity is the
remembered value of the setting, or 0 if the setting was not previously sent.  When the client's 0-RTT value of the setting is 0, the server MAY...

?

Reply:
Thank you for clarifying.  Yes, this sounds better to me.